### PR TITLE
[ISSUE-33] Generate Github data from projects list

### PR DESCRIPTION
**Pull request by lampajr**
**Thank you for submitting this pull request**

fix:
- https://github.com/kiegroup/chain-status/issues/33

**referenced Pull Requests**:

Improvements:
- `definition-file` no more strictly required
- Added `projects` input field for actions
- Get Github data from the provided comma-separated list of projects

### CODE REVIEW

**Review by Ginxo — APPROVED** [review]
LGTM

File: packages/action/src/bin/github-action.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ const { main } = require("./main");
 const start = async () => {
   const args = {
     definitionFile: core.getInput("definition-file"),
+    projects: core.getInput("projects")
+      ? core.getInput("projects").split(",")
+      : undefined,
```
Comment (shubhbapna) [inline packages/action/src/bin/github-action.js:11]:
```suggestion
    projects: core.getInput("projects")?.split(",")
```
I think we can add optional chaining here given that we are working with node >= 14.0

File: packages/action/src/lib/github/main.js [hunk 2]
```diff
@@ -127,16 +127,36 @@ async function mainProjectBranches(project, octokit, baseBranchFilter) {
 async function main(args, outputFolderPath, metadata, skipZero, isDebug) {
   const octokit = createOctokitInstance(args.token);
 
-  logger.info(`Getting projects from definition file ${args.definitionFile}`);
-  const orderedList = await getOrderedListForTree(args.definitionFile, {
-    token: args.token
-  });
+  var orderedList;
```
Comment (shubhbapna) [inline packages/action/src/lib/github/main.js:130]: Change `var` to `let`?

**Review by shubhbapna — CHANGES_REQUESTED** [review]

**Comment by lampajr** [comment]
Using the optional chaining, it looks like ncc fails because it is not able to recognize the token.

```
Error: Module parse failed: Unexpected token (8:40)
File was processed with these loaders:
 * ../../node_modules/@zeit/ncc/dist/ncc/loaders/empty-loader.js
 * ../../node_modules/@zeit/ncc/dist/ncc/loaders/relocate-loader.js
 * ../../node_modules/@zeit/ncc/dist/ncc/loaders/shebang-loader.js

You may need an additional loader to handle the result of these loaders.
|   const args = {
|     definitionFile: core.getInput("definition-file"),
>     projects: core.getInput("projects")?.split(","),
|     outputFolderPath: core.getInput("output-folder-path")
```
Still investigating

File: packages/action/src/lib/github/main.js (hunk 2 above)
Comment (lampajr) [reply packages/action/src/lib/github/main.js]: Changed

**Comment by shubhbapna** [comment]
I would also go with upgrading ncc from `"@zeit/ncc": "^0.22.3"` to `"@vercel/ncc": "^0.31.1"` especially since `@zeit/ncc` is no longer maintained so we would have probably done the upgrade anyways

**Review by shubhbapna — APPROVED** [review]
LGTM

**Comment by Ginxo** [comment]
> Using the optional chaining, it looks like ncc fails because it is not able to recognize the token.
> 
> ```
> Error: Module parse failed: Unexpected token (8:40)
> File was processed with these loaders:
>  * ../../node_modules/@zeit/ncc/dist/ncc/loaders/empty-loader.js
>  * ../../node_modules/@zeit/ncc/dist/ncc/loaders/relocate-loader.js
>  * ../../node_modules/@zeit/ncc/dist/ncc/loaders/shebang-loader.js
> 
> You may need an additional loader to handle the result of these loaders.
> |   const args = {
> |     definitionFile: core.getInput("definition-file"),
> >     projects: core.getInput("projects")?.split(","),
> |     outputFolderPath: core.getInput("output-folder-path")
> ```
> 
> It fails running `ncc build --minify src/bin/github-action.js`, if we want to allow optional chaining we would have to upgrade `"@zeit/ncc": "^0.22.3"` to `"@vercel/ncc": "^0.31.1"` (I'd change to `vercel` since the `zeit` one is no more maintained) otherwise we can keep the ternary operator just for now. @Ginxo @shubhbapna wdyt?

please, go ahead with the `@zeit/ncc` replacement

**Review by Ginxo — CHANGES_REQUESTED** [review]
LGTM, just action patch version should be increased

**Pull request merged**